src/tcgwars/logic/impl/gen4/DiamondPearlPromos.groovy

```diff
@@ -291,248 +188,237 @@ public enum DiamondPearlPromos implements LogicCardInfo {
+            text "40 damage. If the Defending Pokémon tries to attack during your opponent's next turn, your opponent flips a coin. If tails, that attack does nothing."
             energyCost M, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 40
+              sandAttack(thisMove)
             }
           }
           move "Magnet Slash", {
-            text "100 damage. Energy attached to Magnezone."
+            text "100 damage. Discard all [L] Energy attached to Magnezone."
             energyCost L, M, C, C, L
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 100
+              afterDamage {
+                discardAllSelfEnergy(L)
+              }
             }
           }
-
         };
       case DUSKNOIR_DP33:
         return evolution (this, from:"Dusclops", hp:HP130, type:PSYCHIC, retreatCost:2) {
           weakness D, PLUS30
           resistance C, MINUS20
           pokePower "Dark Hide", {
-            text "Once during your turn , you may flip a coin. If heads, look at your opponent�s hand, choose a Pok�mon you find there, and put it on the bottom of his or her deck. This power can�t be used if Dusknoir is affected by a Special Condition."
+            text "Once during your turn (before your attack), you may flip a coin. If heads, look at your opponent's hand, choose a Pokémon you find there, and put it on the bottom of his or her deck. This power can't be used if Dusknoir is affected by a Special Condition."
             actionA {
+              checkNoSPC()
+              checkLastTurn()
+              assert opp.hand.notEmpty() : "Opponent's hand is empty"
+              powerUsed()
+
+              flip {
+                def card = opp.hand.select(min:0, max: 1,"Look at your opponent's hand, choose a Pokémon you find there, and put it on the bottom of his or her deck", cardTypeFilter(POKEMON))
+                if (card.notEmpty()) {
+                  card.moveTo(opp.deck)
+                  shuffleDeck(card, TargetPlayer.OPPONENT)
+                  bc "$thisAbility shuffled $card into deck"
+                }
+              }
             }
           }
           move "Reaper Pulse", {
-            text "70 damage. Move up to 2 damage counters from Dusknoir to 1 of your opponent�s Benched Pok�mon."
+            text "70 damage. Move up to 2 damage counters from Dusknoir to 1 of your opponent's Benched Pokémon."
             energyCost P, P, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 70
+
+              afterDamage {
+                if (self.numberOfDamageCounters && opp.bench) {
+                  def countersToMove = Math.min(self.numberOfDamageCounters, 2)
+                  def target = opp.bench.select("Which target to receive $countersToMove damage counters")
+                  self.damage -= hp(10)
```

Comment:
This only moves one damage counter. Also the player should be allowed to choose to move either 1 or 2 damage counters.